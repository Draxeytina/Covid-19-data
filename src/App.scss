@import "./base.scss";

.filter {
  display: flex;
  justify-content: center;
  padding: $padding;
  margin-bottom: calc($padding/2);
  color: white;
}

@media (min-width: 760px) {
  .profile-grid {
    grid-template-columns: 1fr 1fr;
  }
}
