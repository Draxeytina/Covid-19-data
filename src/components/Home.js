import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { getData } from '../redux/data';
import Data from './Data';
import Filter from './Filter';
import './home.scss';

export default function Home() {
  const dispatch = useDispatch();
  const data = useSelector((state) => state.data);
  const [filtered, setFiltered] = useState([]);
  const [activeCountry, setActiveCountry] = useState('All');

  useEffect(() => {
    if (!data.length) { dispatch(getData()); }
  }, []);

  return (
    <>
      <div className="home">
        Welcome
      </div>

      <Filter
        data={data}
        setFiltered={setFiltered}
        activeCountry={activeCountry}
        setActiveCountry={setActiveCountry}
      />
      <div className="list-container-oval">
        {
          (activeCountry === 'Top World Cases') ? (
            <div>
              <iframe
                title="World Top Data"
                src="https://covid-19.dataflowkit.com/assets/widget/c19-top-widget.html"
                width="90%"
                height="300px"
                frameBorder="0"
              />
            </div>
          ) : (
            <div className="regions-container">
              <ul>
                {filtered.length === 0 && (

                  data.map((data) => (
                    <li key={data.id}>
                      <Data
                        countryName={data.Country_text}
                        totalCases={data['Total Cases_text']}
                        totalDeaths={data['Total Deaths_text']}
                        totalRecoveries={data['Total Recovered_text']}
                      />
                      <Link
                        to={`/covidData/${data.id}`}
                      >
                        <button type="button">View Statistics</button>
                      </Link>
                    </li>
                  ))
                )}
                {filtered.length !== 0 && (

                  filtered.map((data) => (
                    <li key={data.id}>
                      <Data
                        countryName={data.Country_text}
                        totalCases={data['Total Cases_text']}
                        totalDeaths={data['Total Deaths_text']}
                        totalRecoveries={data['Total Recovered_text']}
                      />
                      <Link
                        to={`/covidData/${data.id}`}
                      >
                        <button type="button">View Statistics</button>
                      </Link>
                    </li>
                  ))
                )}
              </ul>
            </div>
          )
        }
      </div>
    </>
  );
}
