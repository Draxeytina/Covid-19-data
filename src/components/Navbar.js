import React, {
  Fragment,
  useEffect,
  useRef,
  useState,
} from 'react';
import {
  Link,
  NavLink,
} from 'react-router-dom';
import { FiX, FiMenu } from 'react-icons/fi';
import './navbar.scss';

const Navbar = () => {
  const [menuMovil, setMenuMovil] = useState(false);
  const [axn, setAxn] = useState(false);
  const cortain = useRef();

  const closeMenu = () => {
    setAxn(false);
  };

  useEffect(() => {
    if (axn) {
      cortain.current.style.display = 'block';
      setMenuMovil(true);
    } else {
      cortain.current.style.display = 'none';
      setMenuMovil(false);
    }
  }, [axn]);

  return (
    <>
      <header>
        <div className="container d-flex">
          <Link to="/" className="logo-link" onClick={() => closeMenu()}>
            <h1>Reddit</h1>
          </Link>

          <div
            className="movil-box"
            role="button"
            tabIndex={-1}
            onClick={() => setAxn(!axn)}
            onKeyUp={() => setAxn(!axn)}
          >
            {
              menuMovil ? (
                <FiX className="movil-box-icon" />
              ) : (
                <FiMenu className="movil-box-icon" />
              )
            }
          </div>

          <nav>
            <ul>
              <li>
                <NavLink to="/home">
                  Home
                </NavLink>
              </li>
              <li>
                <NavLink to="/stock%details">
                  Stock Details
                </NavLink>
              </li>
            </ul>
          </nav>
        </div>
      </header>

      <div className={menuMovil ? 'menu-movil menu-show' : 'menu-movil menu-hide'}>
        <ul>
          <li role="presentation">
            <NavLink to="/home" onClick={() => closeMenu()}>
              Home
            </NavLink>
          </li>
          <li role="presentation">
            <NavLink to="/stock%details" onClick={() => closeMenu()}>
              Stock Details
            </NavLink>
          </li>
        </ul>
      </div>

      <div ref={cortain} className={menuMovil ? 'cortain cortain-show' : 'cortain cortain-hide'}>.</div>
    </>
  );
};

export default Navbar;
