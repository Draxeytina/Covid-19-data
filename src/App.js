import React from 'react';
import { Route, Routes, BrowserRouter } from 'react-router-dom';
import Navbar from './components/Navbar';
import Home from './components/Home';
import LocalData from './components/LocalData';
import Details from './components/Details';
import './App.scss';

function App() {
  return (
    <BrowserRouter>
      <Navbar />
      <div className="main">
        <Routes>
          <Route exact path="/" element={<Home />} />
          <Route exact path="/covidData/:id" element={<Details />} />
          <Route exact path="/local%data" element={<LocalData />} />
          <Route
            path="*"
            element={
              <div><h2>404 Page not found</h2></div>
            }
          />
        </Routes>
      </div>
    </BrowserRouter>
  );
}

export default App;
