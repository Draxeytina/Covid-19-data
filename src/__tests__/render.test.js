import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import App from '../App';
import Home from '../components/Home';
import Navbar from '../components/Navbar';

describe('Test rendering properties', () => {
  test('Display home page', () => {
    const home = render(
      <BrowserRouter>
        <Provider store={store}>
          <Home />
        </Provider>
      </BrowserRouter>,
    );
    expect(home).toMatchSnapshot();
  });
  test('Display the whole app', () => {
    const application = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );
    expect(application).toMatchSnapshot();
  });
  test('Display the nav bar', () => {
    const navBar = render(
      <BrowserRouter>
        <Provider store={store}>
          <Navbar />
        </Provider>
      </BrowserRouter>,
    );
    expect(navBar).toMatchSnapshot();
  });
});
