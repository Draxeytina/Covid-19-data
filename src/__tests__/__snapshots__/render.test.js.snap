// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test rendering properties Display home page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home"
      >
        <h1
          class="welcome-text"
        >
          Welcome
        </h1>
      </div>
      <div
        class="filter"
      >
        Select Region :
         
        <select
          class="categoryInput"
          name="country"
        >
          <option
            value="All"
          >
            All
          </option>
          <option
            value="Southern African Countries"
          >
            Southern African Countries
          </option>
          <option
            value="Northen African Countries"
          >
            Northen African Countries
          </option>
          <option
            value="Western African Countries"
          >
            Western African Countries
          </option>
          <option
            value="Eastern African Countries"
          >
            Eastern African Countries
          </option>
          <option
            value="Middle African Countries"
          >
            Middle African Countries
          </option>
          <option
            value="Top World Cases"
          >
            Top World Cases
          </option>
        </select>
      </div>
      <div
        class="list-container-oval"
      >
        <div>
          <ul
            class="regions-container"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home"
    >
      <h1
        class="welcome-text"
      >
        Welcome
      </h1>
    </div>
    <div
      class="filter"
    >
      Select Region :
       
      <select
        class="categoryInput"
        name="country"
      >
        <option
          value="All"
        >
          All
        </option>
        <option
          value="Southern African Countries"
        >
          Southern African Countries
        </option>
        <option
          value="Northen African Countries"
        >
          Northen African Countries
        </option>
        <option
          value="Western African Countries"
        >
          Western African Countries
        </option>
        <option
          value="Eastern African Countries"
        >
          Eastern African Countries
        </option>
        <option
          value="Middle African Countries"
        >
          Middle African Countries
        </option>
        <option
          value="Top World Cases"
        >
          Top World Cases
        </option>
      </select>
    </div>
    <div
      class="list-container-oval"
    >
      <div>
        <ul
          class="regions-container"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test rendering properties Display the nav bar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <div
          class="container d-flex"
        >
          <a
            class="logo-link"
            href="/"
          >
            <h1>
              Covid 19 Statistics
            </h1>
          </a>
          <div
            class="movil-box"
            role="button"
            tabindex="-1"
          >
            <svg
              class="movil-box-icon"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="3"
                x2="21"
                y1="12"
                y2="12"
              />
              <line
                x1="3"
                x2="21"
                y1="6"
                y2="6"
              />
              <line
                x1="3"
                x2="21"
                y1="18"
                y2="18"
              />
            </svg>
          </div>
          <nav>
            <ul>
              <li>
                <a
                  aria-current="page"
                  class="active"
                  href="/"
                >
                  Home
                </a>
              </li>
              <li>
                <a
                  class=""
                  href="/local%data"
                >
                  Local Statistics
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </header>
      <div
        class="menu-movil menu-hide"
      >
        <ul>
          <li
            role="presentation"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              class=""
              href="/local%data"
            >
              Local Statistics
            </a>
          </li>
        </ul>
      </div>
      <div
        class="visibility visibility-hide"
        style="display: none;"
      >
        .
      </div>
    </div>
  </body>,
  "container": <div>
    <header>
      <div
        class="container d-flex"
      >
        <a
          class="logo-link"
          href="/"
        >
          <h1>
            Covid 19 Statistics
          </h1>
        </a>
        <div
          class="movil-box"
          role="button"
          tabindex="-1"
        >
          <svg
            class="movil-box-icon"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="3"
              x2="21"
              y1="12"
              y2="12"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <line
              x1="3"
              x2="21"
              y1="18"
              y2="18"
            />
          </svg>
        </div>
        <nav>
          <ul>
            <li>
              <a
                aria-current="page"
                class="active"
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                class=""
                href="/local%data"
              >
                Local Statistics
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </header>
    <div
      class="menu-movil menu-hide"
    >
      <ul>
        <li
          role="presentation"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          role="presentation"
        >
          <a
            class=""
            href="/local%data"
          >
            Local Statistics
          </a>
        </li>
      </ul>
    </div>
    <div
      class="visibility visibility-hide"
      style="display: none;"
    >
      .
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test rendering properties Display the whole app 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <div
          class="container d-flex"
        >
          <a
            class="logo-link"
            href="/"
          >
            <h1>
              Covid 19 Statistics
            </h1>
          </a>
          <div
            class="movil-box"
            role="button"
            tabindex="-1"
          >
            <svg
              class="movil-box-icon"
              fill="none"
              height="1em"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="3"
                x2="21"
                y1="12"
                y2="12"
              />
              <line
                x1="3"
                x2="21"
                y1="6"
                y2="6"
              />
              <line
                x1="3"
                x2="21"
                y1="18"
                y2="18"
              />
            </svg>
          </div>
          <nav>
            <ul>
              <li>
                <a
                  aria-current="page"
                  class="active"
                  href="/"
                >
                  Home
                </a>
              </li>
              <li>
                <a
                  class=""
                  href="/local%data"
                >
                  Local Statistics
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </header>
      <div
        class="menu-movil menu-hide"
      >
        <ul>
          <li
            role="presentation"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            role="presentation"
          >
            <a
              class=""
              href="/local%data"
            >
              Local Statistics
            </a>
          </li>
        </ul>
      </div>
      <div
        class="visibility visibility-hide"
        style="display: none;"
      >
        .
      </div>
      <div
        class="main"
      >
        <div
          class="home"
        >
          <h1
            class="welcome-text"
          >
            Welcome
          </h1>
        </div>
        <div
          class="filter"
        >
          Select Region :
           
          <select
            class="categoryInput"
            name="country"
          >
            <option
              value="All"
            >
              All
            </option>
            <option
              value="Southern African Countries"
            >
              Southern African Countries
            </option>
            <option
              value="Northen African Countries"
            >
              Northen African Countries
            </option>
            <option
              value="Western African Countries"
            >
              Western African Countries
            </option>
            <option
              value="Eastern African Countries"
            >
              Eastern African Countries
            </option>
            <option
              value="Middle African Countries"
            >
              Middle African Countries
            </option>
            <option
              value="Top World Cases"
            >
              Top World Cases
            </option>
          </select>
        </div>
        <div
          class="list-container-oval"
        >
          <div>
            <ul
              class="regions-container"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <header>
      <div
        class="container d-flex"
      >
        <a
          class="logo-link"
          href="/"
        >
          <h1>
            Covid 19 Statistics
          </h1>
        </a>
        <div
          class="movil-box"
          role="button"
          tabindex="-1"
        >
          <svg
            class="movil-box-icon"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="3"
              x2="21"
              y1="12"
              y2="12"
            />
            <line
              x1="3"
              x2="21"
              y1="6"
              y2="6"
            />
            <line
              x1="3"
              x2="21"
              y1="18"
              y2="18"
            />
          </svg>
        </div>
        <nav>
          <ul>
            <li>
              <a
                aria-current="page"
                class="active"
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                class=""
                href="/local%data"
              >
                Local Statistics
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </header>
    <div
      class="menu-movil menu-hide"
    >
      <ul>
        <li
          role="presentation"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          role="presentation"
        >
          <a
            class=""
            href="/local%data"
          >
            Local Statistics
          </a>
        </li>
      </ul>
    </div>
    <div
      class="visibility visibility-hide"
      style="display: none;"
    >
      .
    </div>
    <div
      class="main"
    >
      <div
        class="home"
      >
        <h1
          class="welcome-text"
        >
          Welcome
        </h1>
      </div>
      <div
        class="filter"
      >
        Select Region :
         
        <select
          class="categoryInput"
          name="country"
        >
          <option
            value="All"
          >
            All
          </option>
          <option
            value="Southern African Countries"
          >
            Southern African Countries
          </option>
          <option
            value="Northen African Countries"
          >
            Northen African Countries
          </option>
          <option
            value="Western African Countries"
          >
            Western African Countries
          </option>
          <option
            value="Eastern African Countries"
          >
            Eastern African Countries
          </option>
          <option
            value="Middle African Countries"
          >
            Middle African Countries
          </option>
          <option
            value="Top World Cases"
          >
            Top World Cases
          </option>
        </select>
      </div>
      <div
        class="list-container-oval"
      >
        <div>
          <ul
            class="regions-container"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
